torch/csrc/jit/codegen/cuda/manager.cpp

```diff
@@ -37,14 +66,6 @@ c10::Device getDevice(const at::ArrayRef<IValue>& inputs) {
       false, "Could not detect device of inputs to a fusion.");
 }
 
-// CudaFusionManager holds a FusionExecutor and handles all interfacing
-// including compilation and execution.
-//
-// We cache two maps here:
-//   a. string of graph -> kernel_id
-//   b. kernel_id -> FusionExecutor
-//
-// This allows FusionExecutor reuse across nodes;
 class CudaFusionManager {
```

Comment:
it may be nice to have a comment reminding us that CudaFusionManager is not thread safe

Reply:
Good catch!

I think this is potentially to be an issue. We can change the singleton into a `thread_local` instance instead, which would prevent kernel reuse across threads but would be safer.

Thoughts? @tlemo @csarofeen